addons/event/tests/test_event_flow.py

```diff
@@ -108,6 +108,26 @@ def test_event_auto_confirm(self):
         self.assertEqual(test_reg1.state, 'done', 'Event: wrong state of attended registration')
         self.assertEqual(test_event.seats_used, 2, 'Event: incorrect number of attendees after closing registration')
 
+    def test_event_delete_template(self):
+        """ When deleting a template, related lines should be deleted too """
+        template = self.env['mail.template'].create({
+            'name': 'Test template',
+        })
+
+        before_unlink_count = len(self.event_0.event_mail_ids)
+        self.env['event.mail'].create({
+            'event_id': self.event_0.id,
+            'notification_type': 'mail',
+            'interval_unit': 'now',
+            'interval_type': 'before_event',
+            'template_ref': 'mail.template,%i' % template.id,
+        })
+        self.assertEqual(before_unlink_count + 1, len(self.event_0.event_mail_ids))
+
+        template.unlink()
+        after_unlink_count = len(self.event_0.event_mail_ids)
+        self.assertEqual(before_unlink_count, after_unlink_count, "Event mails linked to an unlinked template should be unlinked.")
```

Comment:
For completeness, would be nice to test this with event.mail.type too.

Would go something like:
```py
template = self.event_type_complex.event_type_mail_ids[0].template_ref
event = self.env['event.event'].create({'name': 'event mail template removed', 'event_type_id': self.event_type_complex.id})

self.assertEqual(len(self.event_type_complex.event_type_mail_ids), 2)
self.assertEqual(len(event.event_mail_ids), 2)

template.unlink()

self.assertEqual(len(self.event_type_complex.event_type_mail_ids.exists()), 1)
self.assertEqual(len(event.event_mail_ids.exists()), 1)
```